### PR TITLE
LECTURE: Development Environments Updated and Added

### LINKED ISSUE #279: Add a lecture on development environments
Optionally add a lecture on how to code w/o Jupyter, or with Jupyter lab.

How to work with py files.

Sample dev environments -- VS Code, nvim, etc.

Optionally, how to work with GH.

### PR DESCRIPTION
**Pull request by spulick**
fixes #279 

- incorporated @jstac comments
- included content on using Jupyter notebooks in VS Code

### CODE REVIEW

**Comment by jstac** [comment]
Thanks @orectique .  Could you please look into the failing link check?

**Comment by spulick** [comment]
I've verified that the links flagged by the link checker are all functional. 

It might be that the flagged sites are shutting down automated access.

**Comment by jstac** [comment]
@mmcky or @HumphreyYang , could you please advise?  I think we just ignore the link checker in this case?

**Comment by HengchengZhang** [comment]
Hi @jstac, @mmcky and @HumphreyYang, I checked these links manually and they all should be false positives. My opinion is to add them to the ignore list.

**Comment by HumphreyYang** [comment]
> Hi @jstac, @mmcky and @HumphreyYang, I checked these links manually and they all should be false positives. My opinion is to add them to the ignore list.

Many thanks @HengchengZhang and @jstac. I also think we can have it on since it still works on other links. We can manually check the white list when we have major releases as long as the list is not too long. We can consider disabling it in the future if the list of false positives is getting too long.

**Comment by jstac** [comment]
Thanks @HengchengZhang and @HumphreyYang .  Let's ignore as suggested.

**Comment by jstac** [comment]
@orectique Nice work on this lecture.  Let's meet up and discuss it.  I'll contact you to set up a time.

**Comment by HengchengZhang** [comment]
Hi @jstac I added them to the ignore list and it's working now.

**Comment by spulick** [comment]
Thanks, @jstac.

**Comment by mmcky** [comment]
thanks @orectique -- this looks like it is in-work but let me know if you need a review etc.

**Comment by spulick** [comment]
@mmcky This needs a review, please. I've added new content to the Getting started lecture and revised the workspaces lecture.

**Comment by mmcky** [comment]
Sorry @orectique this slipped through my radar. Can you update on the latest main and then ping me for review. Thanks,.

**Comment by spulick** [comment]
Hi @mmcky,

Thanks for the review. I have merged changes on Main to this branch. There should be no conflicts.

File: .vscode/settings.json
Comment (mmcky) [inline .vscode/settings.json:1]: @orectique can you remove this vscode profile and add to `.gitignore`

File: .ipynb_checkpoints/Demo-checkpoint.ipynb
Comment (mmcky) [inline .ipynb_checkpoints/Demo-checkpoint.ipynb:1]: @orectique can you please remove `.ipynb_checkpoints` and add to `.gitignore`

File: lectures/getting_started.md [hunk 1]
```diff
@@ -382,6 +382,33 @@ Now we `Shift+Enter` to produce this
...
 
+### Debugging Code
+
+Debugging is the process of identifying and removing errors from a program. 
+
+You will spend a lot of time debugging code, so it is important to [learn how to do it effectively](https://www.freecodecamp.org/news/what-is-debugging-how-to-debug-code/).
+
+If you are using a newer version of Jupyter, you should see a bug icon on the right end of the toolbar.
+
+```{figure} /_static/lecture_specific/getting_started/debug.png
```
Comment (mmcky) [inline lectures/getting_started.md:396]: @orectique can we improve the quality i.e. dpi (or make the image smaller)?

File: lectures/workspace.md [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+(workspace)=
+```{raw} jupyter
+<div id="qe-notebook-header" align="right" style="text-align:right;">
+        <a href="https://quantecon.org/" title="quantecon.org">
+                <img style="width:250px;display:inline;" width="250px" src="https://assets.quantecon.org/img/qe-menubar-logo.svg" alt="QuantEcon">
+        </a>
+</div>
+```
+
+# Python Development: Writing Longer Programs
```
Comment (mmcky) [inline lectures/workspace.md]: @orectique I would vote for just `Writing Longer Programs`

File: lectures/workspace.md [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+```{contents} Contents
+:depth: 2
+```
+## Overview
+
+So far, we have explored the use of Jupyter Notebooks in writing and executing Python code. 
+
+While they are efficient and adaptable when working with short pieces of code, Notebooks are not the best choice for longer programs and scripts. 
+
+While Jupyter Notebooks can help execute chunks of code one at a time, scripts allow for long pieces of code to be written and executed in a single go.
```
Comment (mmcky) [inline lectures/workspace.md:33]:
```suggestion
While Jupyter Notebooks can help execute chunks of code one at a time and is well suited to interactive development of used in data science workflows, text files and scripts allow for long pieces of code to be written and executed in a single go. 
```

Comment (mmcky) [inline lectures/workspace.md]:
```suggestion
Jupyter Notebooks are well suited to interactive computing (i.e. data science workflows) and can help execute chunks of code one at a time. Text files and scripts allow for long pieces of code to be written and executed in a single go. 
```

**Comment by mmcky** [comment]
thanks @orectique I reviewed [95de21e](https://github.com/QuantEcon/lecture-python-programming.myst/pull/283/commits/95de21e3deb7e6708ca8472292106aee242024f4) and see they resolved all the outstanding items.  Thank you.

**Comment by mmcky** [comment]
thanks @orectique -- @jstac this is now at review stage.

**Comment by jstac** [comment]
Thanks @mmcky , looks good to me.  It's nicely written.

**Review by jstac — APPROVED** [review]

**Comment by mmcky** [comment]
thanks @jstac and @orectique for these improvements.

**Pull request merged**